test/dash-playlist-loader.test.js

```diff
@@ -30,51 +30,149 @@ QUnit.test('throws if the playlist url is empty or undefined', function(assert)
-  assert.equal(loadedPlaylistStates[1], 'HAVE_METADATA', 'got media second');
+  assert.strictEqual(loadedPlaylist, 2,
+    'fired loadedplaylist once for master playlist and once for' +
+    ' autoselected first child playlist');
+  // this is triggered on the next tick after construction
+  assert.strictEqual(loadedMetadata, 1, 'fired loadedmetadata once');
+  assert.ok(loader.master, 'sets the master playlist');
+  assert.ok(loader.media(), 'sets the media playlist');
+  assert.strictEqual(this.requests.length, 0, 'no further requests are needed');
+
+  const childLoader = new DashPlaylistLoader(loader.master.playlists[0], this.fakeHls, false, loader);
+
+  childLoader.on('loadedplaylist', function() {
+    loadedPlaylist++;
+  });
+  childLoader.on('loadedmetadata', function() {
+    loadedMetadata++;
+  });
+
+  childLoader.load();
+  this.clock.tick(1);
+
+  // actually, loadedplaylist is triggered 4 times at this point,
+  // but the 3rd is on childLoader's creation so it's not observable
+  assert.strictEqual(loadedPlaylist, 3, 'fired loadedplaylist thrice');
+  assert.strictEqual(loadedMetadata, 2, 'fired loadedmetadata twice');
+  assert.strictEqual(loader.state, 'HAVE_METADATA', 'the state is correct');
+});
+
+QUnit.test('jumps to HAVE_METADATA when initialized with a live media playlist',
+  function(assert) {
+    const loader = new DashPlaylistLoader('dash-live.mpd', this.fakeHls);
+    let loadedmetadatas = 0;
+
+    loader.load();
+    standardXHRResponse(this.requests.shift());
+
+    assert.strictEqual(loadedmetadatas, 0, 'fired no loadedmetadata on same tick');
+    assert.ok(loader.master, 'infers a master playlist');
+    assert.ok(loader.media(), 'sets the media playlist');
+    assert.ok(loader.media().attributes, 'sets the media playlist attributes');
+
+    this.clock.tick(1);
+
+    assert.strictEqual(loader.state, 'HAVE_METADATA', 'the state is correct');
+  });
+
+QUnit.test('returns to HAVE_METADATA after refreshing the playlist', function(assert) {
+  let loader = new DashPlaylistLoader('dash-live.mpd', this.fakeHls);
+
+  loader.load();
+
+  standardXHRResponse(this.requests.shift());
+  // 10s, one target duration
+  this.clock.tick(10 * 1000);
+
+  assert.strictEqual(loader.state, 'HAVE_METADATA', 'the state is correct');
+});
+
+QUnit.test('triggers an event when the active media changes', function(assert) {
+  let loader = new DashPlaylistLoader('dash.mpd', this.fakeHls);
+  let mediaChanges = 0;
+  let mediaChangings = 0;
+
+  loader.on('mediachange', function() {
+    mediaChanges++;
+  });
+  loader.on('mediachanging', function() {
+    mediaChangings++;
+  });
+
+  loader.load();
+  standardXHRResponse(this.requests.shift());
+  this.clock.tick(1);
+
+  assert.strictEqual(mediaChangings, 0, 'initial selection is not a media changing');
+  assert.strictEqual(mediaChanges, 0, 'initial selection is not a media change');
+
+  loader.media(loader.master.playlists[1]);
```

Comment:
This call would be considered "initial selection" and should result in `0` `mediachanging/mediachange` events. The rest of the assertions would cascade a `-1` to account for this